Documentation/ApiOverview/Http/Index.rst

```diff
@@ -1,103 +1,109 @@
-already used within TYPO3.
+The PHP library "Guzzle" is available in TYPO3 as a feature-rich solution for
+creating HTTP requests based on the PSR-7 interfaces.
 
-Guzzle auto-detects available underlying adapters available on the system, like cURL or
-stream wrappers and chooses the best solution for the system.
+Guzzle automatically detects the underlying adapters available on the system,
+such as cURL or stream wrappers, and chooses the best solution for the system.
 
-A TYPO3-specific PHP class called :php:`\TYPO3\CMS\Core\Http\RequestFactory` has been added as
-a simplified wrapper to access Guzzle clients.
+A TYPO3-specific PHP class named :php:`TYPO3\CMS\Core\Http\RequestFactory` is
+present as a simplified wrapper for accessing Guzzle clients.
 
-All options available under :php:`$GLOBALS['TYPO3_CONF_VARS'][HTTP]` are automatically applied to the Guzzle
-clients when using the :php:`RequestFactory` class. The options are a subset to the available options
-on Guzzle (https://docs.guzzlephp.org/en/latest/request-options.html) but can further be extended.
+All options available under :php:`$GLOBALS['TYPO3_CONF_VARS']['HTTP']` are
+automatically applied to the Guzzle clients when using the :php:`RequestFactory`
+class. The options are a subset of the available options from Guzzle, but can be
+extended.
 
-Existing :php:`$GLOBALS['TYPO3_CONF_VARS'][HTTP]` options have been removed and/or migrated to the
-new Guzzle-compliant options.
+..  seealso::
+    -   :ref:`typo3ConfVars_http`
+    -   `Guzzle request options <https://docs.guzzlephp.org/en/stable/request-options.html>`__
+    -   `Full documentation for Guzzle <https://docs.guzzlephp.org/en/stable/>`__
 
-A full documentation for Guzzle can be found at https://docs.guzzlephp.org/en/latest/.
+Although Guzzle can handle Promises/A+ and asynchronous requests, it currently
+serves as a drop-in replacement for the previous mixed options and
+implementations within :php:`GeneralUtility::getUrl()` and a PSR-7-based API for
+HTTP requests.
 
-Although Guzzle can handle Promises/A+ and asynchronous requests, it currently acts as
-a drop-in replacement for the previous mixed options and implementations within
-:php:`GeneralUtility::getUrl()` and a PSR-7-based API for HTTP requests.
-
-The existing TYPO3-specific wrapper :php:`GeneralUtility::getUrl()` now uses Guzzle under the hood
-automatically for remote files, removing the need to configure settings based on certain
-implementations like stream wrappers or cURL directly.
+The TYPO3-specific wrapper :php:`GeneralUtility::getUrl()` uses Guzzle for
+remote files, eliminating the need to directly configure settings based on
+specific implementations such as stream wrappers or cURL.
 
 .. index:: HTTP request; RequestFactory
 .. _http-basic:
 
 Basic usage
 ===========
 
-The `RequestFactory` class can be used like this:
+The :php:`RequestFactory` class can be used like this:
+
+..  code-block:: php
```

Comment:
Would it be possible to make a real, working example in EXT:examples or some such? Pseudo Code has the tendency to degenerate over time.

Reply:
Yes, thought also about it. Created an issue for that: https://github.com/TYPO3-Documentation/t3docs-examples/issues/108
Would like to get this merged as the example is wrong and I am going on vacation in a week. Before, I also want to adjust the example vor v12.0 with PHP 8.1 (constructor property promotion/readonly).

Reply:
Ok, we can do that in a follow up